.circleci/config.yml

```diff
@@ -83,6 +83,19 @@ jobs:
           name: check optimized images
           command: make lint-assets
 
+  check-javascript-syntax:
+    docker:
+      - image: circleci/ruby:2.6-node-browsers
+    steps:
+      - checkout
+      - install-dependencies
+      - run:
+          name: build javascript
+          command: npm run build-js
```

Comment:
We have a lot of parallel build steps now.

Our JS build is fairly fast, so maybe not worth it here, but this is duplicated from our `build-site` command. Maybe would it be worth having a multi-phase job where we build the site once in `build-site` and then save the workspace and load that workspace in all the jobs that run the various checks? 


So it would look like:
```
[build site, cache artifact] ->
  - run JS lint
  - run JS specs
  - run ruby specs on built site, etc etc
```

Reply:
Some example of workspace saving across steps in the old handbook: https://github.com/18F/identity-handbook-private/blob/master/.circleci/config.yml#L47-L59

Reply:
Yeah, that sounds really great. I've tried incorporating this for "base" tests, as well as the external links checking which has a similar need. The [timings](https://app.circleci.com/pipelines/github/18F/identity-site?branch=aduth-try-es5-safe) aren't quite as promising as I'd have hoped, and in-fact the builds take slightly longer than before I'd incorporated it.

Still, it seems like a good idea in theory, and might help some future scalability, so I'd be fine to keep the changes 🤷